<?xml version="1.0"?>
<robot name="241">
  <!-- 색상 정의 -->
  <material name="blue">
    <color rgba="0.0 0.0 1.0 1.0"/>
  </material>
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="gray">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="yellow">
    <color rgba="1.0 1.0 0.0 1.0"/>
  </material>
  <material name="red">
    <color rgba="1.0 0.0 0.0 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 1.0 0.0 1.0"/>
  </material>

  <!-- Base Link -->
  <link name="base_link">
    <visual>
        <origin xyz="0 0 0"/>
    </visual>
  </link>

  <!-- Left Hull -->
  <link name="left_hull">
    <visual>
      <origin xyz="0.0 -0.4 0"/>
      <geometry>
        <box size="1.2 0.175 0.25"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>

  <joint name="left_hull_joint" type="fixed">
    <parent link="base_link"/>
    <child link="left_hull"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Right Hull -->
  <link name="right_hull">
    <visual>
      <origin xyz="0.0 0.4 0"/>
      <geometry>
        <box size="1.2 0.175 0.25"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_hull_joint" type="fixed">
    <parent link="base_link"/>
    <child link="right_hull"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Servo 1 -->
  <link name="servo1">
    <visual>
      <origin xyz="-0.5 0.4 -0.125"/>
      <geometry>
        <cylinder radius="0.025" length="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <joint name="servo1_joint" type="continuous">
    <parent link="left_hull"/>
    <child link="servo1"/>
    <origin xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <!-- Servo 2 -->
  <link name="servo2">
    <visual>
      <origin xyz="-0.5 -0.4 -0.125"/>
      <geometry>
        <cylinder radius="0.025" length="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <joint name="servo2_joint" type="continuous">
    <parent link="right_hull"/>
    <child link="servo2"/>
    <origin xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <!-- Thruster 1 -->
  <link name="thruster1">
    <visual>
      <origin xyz="-0.5 0.4 0.3" rpy="0 1.5708 0"/> 
      <geometry>
        <cylinder radius="0.05" length="0.05"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>

  <joint name="thruster1_joint" type="fixed">
    <parent link="servo1"/>
    <child link="thruster1"/>
    <origin xyz="0 0 0 "/>
  </joint>

  <!-- Thruster 2 -->
  <link name="thruster2">
    <visual>
      <origin xyz="-0.5 -0.4 0.3" rpy="0 1.5708 0"/>
      <geometry>
        <cylinder radius="0.05" length="0.05"/>
      </geometry>
      <material name="gray"/>
    </visual>
  </link>

  <joint name="thruster2_joint" type="fixed">
    <parent link="servo2"/>
    <child link="thruster2"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Lidar -->
  <link name="velodyne">
    <visual>
      <origin xyz="0.1 0 0.3"/>
      <geometry>
        <cylinder radius="0.1" length="0.05"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <joint name="velodyne_joint" type="fixed">
    <parent link="base_link"/>
    <child link="velodyne"/>
    <origin xyz="0.1 0 0.3"/>
  </joint>

</robot>
